src/materials/ConcreteExpansionMicrocrackingDamage.C

```diff
@@ -0,0 +1,167 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ConcreteExpansionMicrocrackingDamage.h"
+#include "ElasticityTensorTools.h"
+
+registerMooseObject("BlackBearApp", ConcreteExpansionMicrocrackingDamage);
+
+template <>
+InputParameters
+validParams<ConcreteExpansionMicrocrackingDamage>()
+{
+  InputParameters params = validParams<ScalarDamageBase>();
+  params.addClassDescription("Scalar damage model based on extent of internal expansion");
+
+  params.addRequiredParam<MaterialPropertyName>(
+      "eigenstrain_name", "name of the eigenstrain driving the microcracking damage process");
+
+  params.addParam<bool>("use_isotropic_expansion",
+                        true,
+                        "indicates whether the model assumes an isotropic expansion (true) or "
+                        "decomposes the expansion alongside the principal direction (false)");
+
+  params.addParam<bool>("use_stress_control",
+                        true,
+                        "indicates whether the damage is affected by the current stress state");
+
+  params.addParam<Real>("correction_factor",
+                        1.0,
+                        "correction factor by which the eigenstrain is multiplied before "
+                        "evaluating the damage");
+
+  params.addRequiredRangeCheckedParam<Real>(
+      "microcracking_strain_init",
+      "microcracking_strain_init > 0",
+      "linear strain at which the microcracking initiates (in [m/m])");
+
+  params.addRequiredRangeCheckedParam<Real>(
+      "microcracking_strain_branch",
+      "microcracking_strain_branch > 0",
+      "parameter controlling the rate at which the microcracking increases (in [m/m])");
+
+  params.addRangeCheckedParam<Real>(
+      "expansion_stress_limit",
```

Comment:
Does this actually need to be greater than some tolerance? It looks like you could get close to a divide by zero condition.

Reply:
This is generally in the order of magnitude of several MPa. We could set it > 1 for safety?